internal/config/config.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/project"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/yaml"
+)
+
+const DefaultPath = "PROJECT"
+
+// Config is the unmarshalled representation of the configuration file
+type Config struct {
+	// Version is the project version, defaults to "1" (backwards compatibility)
+	Version string `json:"version,omitempty"`
+
+	// Domain is the domain associated with the project and used for API groups
+	Domain string `json:"domain,omitempty"`
+
+	// Repo is the go package name of the project root
+	Repo string `json:"repo,omitempty"`
+
+	// Resources tracks scaffolded resources in the project
+	// This info is tracked only in project with version 2
+	Resources []Resource `json:"resources,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+// Check verifies that the configuration file exists
+func Check(path string) (bool, error) {
+	// Look up the file
+	_, err := os.Stat(path)
+
+	// If we could find it the file exists
+	if err == nil {
+		return true, nil
+	}
+
+	// Not existing and different errors are differentiated
+	if os.IsNotExist(err) {
+		err = nil
+	}
+	return false, err
+}
+
+func LoadFrom(path string) (*Config, error) {
```

Comment:
```suggestion
func loadFrom(path string) (*Config, error) {
```

Reply:
`LoadFrom` should still be exported as the configuration file may be placed elsewhere. For example for tests inside `testdata` dir.

Also, despite we only support calling `kubebuilder` commands from the root directory right now, in the future we may allow to do it from other places, for example with a `--dir="my/path"` flag that sets the place where the project is located. That would also require `LoadFrom`.

Also, exporting `LoadFrom` is not increasing the attack surface as this package is under an `internal` directory, so we will only be able to call it from `kubebuilder`.

Reply:
> LoadFrom should still be exported as the configuration file may be placed elsewhere. For example for tests inside testdata dir.

Could you please add a link for an example? 

> Also, despite we only support calling kubebuilder commands from the root directory right now, in the future we may allow to do it from other places, for example with a --dir="my/path" flag that sets the place where the project is located. That would also require LoadFrom.

Its is premature optimization and because of this, it probably would have not the best design/solution now. Do you know if the project has a flag `--dir="my/path"` to check the config is part of its roadmap? Do you know if in the future instead of having a flag it may support diff types of project where each type has its own default config file for example? 

Also, If we move forward for any direction in the future we can easily implement something to use the `loadFrom/save/edit(path string)` after a good discussion over the best solution to attend its needs. Now, I think we can just avoid let the project file be read/edit/save in any path the middle of the code in order to minimize risks such as, for example, end up with more than one project file in the project. Is it make sense?

Reply:
> Could you please add a link for an example?

https://github.com/Adirio/kubebuilder/blob/6474bc287b76bc2d8e76e51016db0403828abd12/pkg/scaffold/scaffold.go#L135 is an example of how you do need to `LoadFrom` other places. (the link may break with the next force-push, it points to the `Execute` function of the `Scaffold` struct in `pgk/scaffold/scaffold.go`)

The `--dir` flag comment was just an extra advantage, the main point is it is still needed for `Scaffold.Execute`. The only thing that I did taking into account possible future directions i not using the words "project" and "file" as they are not descriptive of what the struct holds, but that is not premature optimization as it is just a naming design decission.

I however do like what you say about not allowing to store it in a different place by mistake. I will make a `Save()` method that uses the same path where it was loaded from, and a `CopyTo(path string)` that makes clear that we are creating a new copy.